### PR TITLE
fix: stop the staleness warning promising a rebuild it does not perform (#1649)

### LINKED ISSUE #1649: The staleness warning's rebuild remedy leaves stale edges behind: 12 of 82 survive on gin
The staleness warning tells a user to delete `.cgr-hash-cache.json`, `.cgr-dir-mtimes.json` and `.cgr-parser-fingerprint` and re-index, promising "every file is then treated as new". That re-parse does not fully reconcile relationships: edges the OLD parser inputs emitted can survive it, and the graph keeps answering with them.

Measured on gin-gonic/gin across the #1641 fix, which removes spurious `INSTANTIATES` edges:

| graph | INSTANTIATES |
|---|---|
| indexed before the fix | 82 |
| after the documented remedy (delete 3 caches, re-index with the fix) | **12** |
| a fresh copy of the same source indexed with the fix | **0** |

All 12 survivors are absent from the fresh copy, so every one is stale.

**The partial result is the dangerous part.** 70 of 82 disappearing looks exactly like the remedy working. A user has no signal that 12 remain, and the relation is now a mixture of current and superseded facts rather than being obviously wrong.

Found while building the whole-project controls for #1641.

## Reproduction

```bash
git clone --depth 1 https://github.com/gin-gonic/gin.git
# index with a build that predates the fix
cgr start --repo-path ./gin --update-graph

# apply the #1641 fix, then follow the warning verbatim
rm ./gin/.cgr-hash-cache.json ./gin/.cgr-dir-mtimes.json ./gin/.cgr-parser-fingerprint
cgr start --repo-path ./gin --update-graph
```

```cypher
MATCH (a)-[:INSTANTIATES]->(b) WHERE a.qualified_name STARTS WITH 'gin__'
RETURN count(*)
-- 12, against 0 for a fresh copy of identical source
```

## The survivors

```
context.Context.ClientIP            -> context.Context
context.Context.Deadline            -> context.Context
context.Context.Done                -> context.Context
context.Context.Err                 -> context.Context
context.Context.Value               -> context.Context
context.Context.hasRequestContext   -> context.Context
context_test.TestContextClientIP    -> context.Context
context_test.TestContextCopyCopiesErrors -> errors.Error   (x4)
errors.Error.Error                  -> errors.Error
```

Every source file above was re-parsed by the second run — the caches were gone, so nothing was skipped. The edges survived the re-parse itself.

## What the remedy should do

A rebuild after a fingerprint change must reconcile **edges**, not only nodes — or, if that is not intended, the warning must say "re-create the project" rather than describing a cache deletion as a rebuild.

The current text is specific and confident about what it achieves ("every file is then treated as new"), which is what makes the partial result misleading. It already carries a caveat for a different case ("does not remove nodes the old ones left behind... run `cgr start --clean --update-graph` instead"), so the shape of a correct warning exists; it is the edge case that is unaccounted for.

Note that `--clean` is not a usable workaround here: it deletes **every project** in a shared database, not just the one being rebuilt, as the warning itself says. For the #1641 controls I indexed copies as new projects instead, which needs no deletion at all.

## Not assumed

I have not established which write path leaves these 12 behind, nor why these particular callers rather than the other 70. The four duplicate `TestContextCopyCopiesErrors -> errors.Error` rows suggest per-site edges (each carries its own line/col) may be reconciled differently from single-site ones, but I did not confirm that and it should not be read as the cause.

## Environment

```
cgr             0.0.842 plus the #1641 fix
repo            vitali87/code-graph-rag
target          gin-gonic/gin at depth-1 HEAD, 99 .go
graph           Memgraph, projects gin__1cf03975 (remedy) and gin-after__cfa96887 (fresh)
```

Counts taken after each indexing process exited.

### PR DESCRIPTION
**Pull request by vitali87**
Closes #1649.

## The problem

The staleness warning told users to delete three cache files and re-index, promising **"every file is then treated as new"**. Measured on gin-gonic/gin across the #1641 fix:

| graph | `INSTANTIATES` |
|---|---|
| indexed before the fix | 82 |
| after the documented remedy | **12** |
| a fresh copy of identical source | **0** |

All 12 survivors were stale. **The partial result is the dangerous part**: 70 of 82 disappearing looks exactly like the remedy working, and the user gets no signal that 12 remain.

## Which option, and why

The issue offers a rebuild that reconciles edges, **or** a warning that stops describing a cache deletion as a rebuild. Taking the second, and the mechanism says why the first is a much larger change than it looks:

```cypher
MATCH (m:Module {path: $path}) ...
OPTIONAL MATCH (m)-[:DEFINES|DEFINES_METHOD|CONTAINS_SECTION*0..]->(c)
DETACH DELETE m, c
```

A re-parse removes the module and what it **DEFINES**. Every survivor in the issue points at `context.Context` or `errors.Error` — inferred or external nodes that no re-parsed module defines, so they sit outside the deleted subtree and their edges outlive it. Reconciling those means deleting by *provenance* rather than by containment, which is a design change to how edges are attributed, not a fix to this message.

## The change

The warning now says what the remedy actually achieves: a partial rebuild, that a re-parse only removes what the re-parsed module DEFINES, that nodes **and edges** outside that subtree survive, the measured 12-of-82 result, and two routes that do work — indexing the source as a new project (which needs no deletion at all) or `--clean`.

The pre-existing caveat that `--clean` deletes **every** project in a shared database is kept, and a test guards it: a rewrite that dropped it to make room for the new caveat would trade one hazard for a worse one.

## Tests

`codebase_rag/tests/test_staleness_warning_is_honest.py`, 6 tests asserting **properties** rather than exact wording — the text may be reworded, but it must not go back to promising a complete rebuild, and must keep naming the edge case, the mechanism, a working route, and the `--clean` hazard.

Verified against the old text: three fail (the promise, the mechanism, the working route) and three pass — the three that pass are exactly the pre-existing caveats this rewrite preserves, which is what shows it corrected the message without weakening what was already right.

Regression: 102 passed across the staleness/fingerprint/logs test files.

## Summary by CodeRabbit

* **Documentation**
  * Updated parser warnings to clarify that partial rebuilds may leave stale edges.
  * Added clearer recovery guidance, including indexing under a new project or running a clean graph update.
  * Retained information about shared-database cleanup and relevant cache files.

* **Tests**
  * Added coverage confirming warnings describe rebuild limitations, recovery options, cleanup considerations, and cache-file details.

### CODE REVIEW

**Comment by vitali87** [comment]
@greptileai review

File: codebase_rag/logs.py [hunk 1]
```diff
@@ -983,12 +983,19 @@
...
+    "are untouched. That is a partial rebuild, not a clean one: it adds what "
+    "the new inputs emit, but a re-parse only removes what the re-parsed "
+    "module DEFINES, so nodes and edges the old inputs left outside that "
+    "subtree survive it. Measured on one repository, 12 of 82 stale edges "
+    "outlived the cache deletion because their other end was an inferred or "
+    "external node the re-parse never touched (issue #1649), and a partial "
+    "result looks exactly like the remedy working. When names or node kinds "
+    "changed, or when superseded EDGES must go rather than new ones merely "
+    "being added, the only reliable rebuild is to index the source as a NEW "
+    "project, or to run 'cgr start --clean --update-graph'. Note '--clean' "
```
[the remark this answers is not in the document]
Comment (vitali87) [reply codebase_rag/logs.py]: Fair point, and fixed in dc4a1d15. A route the reader cannot follow is not a route, and the obvious guess - re-running the same command - reproduces the problem the warning is about.

The warning now gives the command:

  cgr start --repo-path <path> --project-name <a-fresh-name> --update-graph

and says what it buys: it deletes nothing, and leaves the old project in place to compare against, which is what makes it safer than --clean on a shared store.

Verified --project-name is actually accepted by `start` rather than assuming it from another subcommand, and the test asserts it against the CLI source rather than against a copy of the string - so if the flag is ever renamed, the test fails instead of the advice silently going stale.

**Comment by vitali87** [comment]
@greptileai review

Requesting a re-review at the current head `dc4a1d156fe6`. Every previous review on this PR anchored to the commit before my next push, so the verdict has never covered the final state; no further pushes are planned, so this one should settle at head.
